x-pack/plugins/security_solution/common/endpoint/types/index.ts

```diff
@@ -492,9 +492,11 @@ export type HostInfo = Immutable<{
   };
 }>;
 
-// HostMetadataDetails is now just HostMetadata
-// HostDetails is also just HostMetadata
-export type HostMetadata = Immutable<{
+// Host metadata document streamed up to ES by the Endpoint running on host machines.
+// NOTE:  `HostMetadata` type is the original and defined as Immutable. If needing to
+//        work with metadata that is not mutable, use `HostMetadataInterface`
+export type HostMetadata = Immutable<HostMetadataInterface>;
```

Comment:
Overtime, we will likely remove the `Immutable<>` from several of our types. It was used mainly when we had Redux to ensure devs did not accidently mutated data and impact other consumers of the store